### DISCUSSION TITLE
Best way to invalidate entities in graphcache from "outside"?

### DISCUSSION THREAD

**Discussion by cristianpavel-xrite** [urql-graphql/urql#2283] [Q&A] [answered]
I have a type of data on the server that is updated through a different API (has to go through a different service).
What's the best way I can signal to the graphcache that it should re-fetch that data?

At the moment I'm running the query from the client directly after I know I updated the data:
```js
import { useClient } from 'urql';
...
const client = useClient();
...
await someRestApi.updateThingOnTheServer();
client
  .query(ThingDataDocument, undefined, { requestPolicy: 'network-only' })
  .toPromise(); // have to call .toPromise to trigger the query, the stream is lazy I think
```
I've also had success with `reExecuteQuery` from `useQuery`, but that feels wrong since I don't care about the data at that point, I just want to tell the cache to update. I also want to avoid unnecessarily mixing declarative and imperative logic.
I think a subscription would be a better solution here, but the server isn't set up for it yet.

Is there a better way to do this?
I'd prefer something like a local-only mutation that only works on the cache but doesn't actually send anything to the server, plus an updater that does the invalidation, but I haven't had any success with that and I'm not sure it's even possible.

I'm happy to get a "no" answer or a frame challenge, I just want to make sure I didn't miss anything obvious. Thanks!

**Comment by JoviDeCroock** [comment, marked as answer]
Currently we have no solution for that, we have been thinking about it in https://github.com/FormidableLabs/urql/pull/2076 but still think it could introduce a lot of footguns.

As you mention a manual refetch or subscription is a good solution, however some kind off control-flow exchange that allows you to simulate a mutation response would probably be a good thing to implement as well.
